spec/classes/mail_on_failure_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'spec_helper'
+
+describe 'systemd::mail_on_failure' do
+  context 'supported operating systems' do
+    on_supported_os.each do |os, facts|
+      context "on #{os} with default params" do
+        let(:facts) { facts }
+
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to create_class('systemd::mail_on_failure') }
+        it {
+          is_expected.to contain_file('/usr/local/bin/systemd-email').with(
+            source: 'puppet:///modules/systemd/systemd-email',
+            owner: 'root',
+            group: 'systemd-journal',
+            mode: '0750',
+          )
+        }
+        it {
+          is_expected.to contain_systemd__unit_file('status_email_root@.service').with(
+            # for some reason facts['networking']['fqdn'] is not valid
+            content: %r{root@#{facts['fqdn']}},
+            require: 'File[/usr/local/bin/systemd-email]',
```

Comment:
The best way to check dependencies is to use `.that_requires('File[/usr/local/bin/systemd-email]')` since it checks the compiled catalog rather than the property. That means it also checks transitive dependencies.

Reply:
ok, didn't (yet) know that one.